app/src/components/Courses/PersonalInsights.tsx

```diff
@@ -19,52 +25,89 @@ type PersonalInsightsProps = {
 const PersonalInsights = (props: PersonalInsightsProps) => {
   const courses = props.courses ?? [];
 
+  // Gets the total number of subscribers from all courses written by the content creator
+  function getAllSubscribers() {
+    let subscribers = 0;
+    courses.forEach(course => {
+      subscribers += course.numOfSubscriptions;
+    });
+    return subscribers;
+  }
+
+  // Gets the total number of courses written by the content creator
+  function getNumberOfCourses() {
+    return courses.length;
+  }
+
+  // Gets the average rating of all courses written by the content creator
+  function getAverageRating() {
+    let avgRating = 0;
+    courses.forEach(course => {
+      avgRating += course.rating;
+    });
+    return avgRating / courses.length;
+  }
+
+  // Gets the number of courses with a specific status
+  function getCourseCountWithStatus(status: string) {
+    let count = 0;
+    courses.forEach(course => {
+      if (course.status === status) {
+        count++;
+      }
+    });
+    return count;
+  }
+
   // TODO: Implement functionality for user info fields
   return (
     <div className='w-full h-full'>
       {/* Welcome message */}
-      <h2 className='text-xl font-bold'>Olá User Name</h2>
+      <h2 className='text-xl font-bold'>Olá {userInfo.name}</h2>
       {/* Progress section (stats) */}
-      <div className='border-y-[1px] border-grayMedium my-4 py-8 '>
-        <p className='font-bold'>Progresso</p>
+      <div className='border-y-[1px] border-grayMedium my-4 py-4'>
         <div className='grid grid-cols-4 lg:grid-cols-1 gap-y-4'>
           {/* Show stats if the user has > 0 courses */}
           {courses.length ? <>
-          <div>
-            {/* Total courses */}
-            <p>Total cursos</p>
-            <p className='font-bold'>8</p>
-          </div>
-          <div>
-            {/* Total students */}
-            <p>Total alunos</p>
-            <p className='font-bold'>167</p>
-          </div>
-          <div>
-            {/* Total certificates issued */}
```

Comment:
Is total certificates not needed or will it just be implemented another time

Reply:
They will be changed, instead we'll show the percentage of students that have completed the user's courses 👍